rest/context.go

```diff
@@ -325,6 +351,60 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) buildURL(fullPath bool, resourceName string,
+	methodName string, pairs ...string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("unable to build URL for resource name %q: no request available.",
+			resourceName)
+	}
+
+	routeName := resourceName + ":" + methodName
+	route := ctx.router.Get(routeName).Host(r.Host)
+
+	var builder func(pairs ...string) (*url.URL, error)
```

Comment:
That's passed straight through to [gorilla's Route.URL() method](http://www.gorillatoolkit.org/pkg/mux#Route.URL).

I'm not in love with it either, but I'm not sure if the alternatives are any better in Go land. I considered a map, which would be a decent solution if not for the `map[string]string{}` line noise necessary on usage.

Keyword args sure would be nice...